include/fluent-bit/flb_aws_util.h

```diff
@@ -160,6 +160,8 @@ int flb_aws_is_auth_error(char *payload, size_t payload_size);
 
 int flb_read_file(const char *path, char **out_buf, size_t *out_size);
 
+//* Constructs S3 object key as per the format. */
+flb_sds_t get_s3_key(const char *format, time_t time, char *tag, char *tag_delimiter);
```

Comment:
Sorry, I should have said this before, but all functions in header files in the core must be prefixed with `flb`. So this has to be `flb_s3_key`